src/main/java/com/fexl/circumnavigate/mixin/chunkHandle/ChunkTrackerMixin.java

```diff
@@ -23,6 +23,103 @@ protected ChunkTrackerMixin(int firstQueuedLevel, int width, int height) {
+//
+//		return getComputedWrappedLevel(chunkPos, excludedSourcePos, level);
+//	}
+//
+//	@WrapOperation(method = "getComputedLevel", at = @At(value = "INVOKE", target = "Lnet/minecraft/server/level/ChunkTracker;computeLevelFromNeighbor(JJI)I"))
+//	private int getComputedLevel(ChunkTracker instance, long startPos, long endPos, int startLevel, Operation<Integer> original, @Local(name = "excludedSourcePos") long excludedSourcePos, @Local(argsOnly = true) int level) {
+//		if (startPos == ChunkPos.INVALID_CHUNK_POS) {
+//			return original.call(instance, startPos, endPos, startLevel);
+//		}
+//
+//		ChunkPos pos = new ChunkPos(startPos);
+//		WorldTransformer transformer = TransformerRequests.chunkCacheLevel.getTransformer();
+//
+//		// Return original function if chunk won't be wrapped
+//		boolean isOverBounds = transformer.isChunkOverBounds(pos);
+//		if (!isOverBounds) {
+//			return original.call(instance, startPos, endPos, startLevel);
+//		}
+//
+//		return getComputedWrappedLevel(pos, excludedSourcePos, level);
+//	}
+//
+//	/*
+//	* Custom getComputedLevel method for wrapped chunks. This is essentially what fixes #10.
+//	* */
+//	@Unique
+//	private int getComputedWrappedLevel(ChunkPos chunkPos, long excludedSourcePos, int level) {
+//		WorldTransformer transformer = TransformerRequests.chunkCacheLevel.getTransformer();
+//		ChunkPos wrappedChunkPos = transformer.translateChunkToBounds(chunkPos);
+//		long wrappedPos = wrappedChunkPos.toLong();
+//
+//		int j = wrappedChunkPos.x;
+//		int k = wrappedChunkPos.z;
+//
+//		for (int l = -1; l <= 1; l++) {
+//			for (int m = -1; m <= 1; m++) {
+//				ChunkPos neighborChunkPos = new ChunkPos(j + l, k + m);
+//				long neighborPos = neighborChunkPos.toLong();
+//				ChunkPos wrappedNeighborChunkPos = transformer.translateChunkToBounds(neighborChunkPos);
+//				long wrappedNeighborPos = wrappedNeighborChunkPos.toLong();
+//
+//				if (wrappedNeighborPos == wrappedPos) {
+//					wrappedNeighborPos = ChunkPos.INVALID_CHUNK_POS;
+//				}
+//
+//				if (neighborPos == wrappedPos) {
+//					neighborPos = ChunkPos.INVALID_CHUNK_POS;
+//				}
+//
+//				if (wrappedNeighborPos != excludedSourcePos) {
+//					int levelFromWrappedNeighbor = computeLevelFromNeighbor(wrappedNeighborPos, wrappedPos, getLevel(wrappedNeighborPos));
+//					if (level > levelFromWrappedNeighbor) {
+//						level = levelFromWrappedNeighbor;
+//					} else {
+//						int levelFromNeighbor = computeLevelFromNeighbor(neighborPos, wrappedPos, getLevel(neighborPos));
+//						if (level > levelFromNeighbor) {
+//							level = levelFromNeighbor;
+//						}
+//					}
+//
+//					if (level == 0) {
+//						return level;
+//					}
+//				}
+//			}
+//		}
+//
+//		if (level >= ChunkLevel.MAX_LEVEL) {
+//            Circumnavigate.LOGGER.warn("Level: {} ChunkPos: {} WrappedChunkPos: {}", level, chunkPos, wrappedChunkPos);
+//		}
+//
+//
+//		// make sure to not go over the max level
+//		return Math.min(ChunkLevel.MAX_LEVEL, level);
+//	}
```

Comment:
All of this commented code can replace mixin below, but results in the same issue anyway